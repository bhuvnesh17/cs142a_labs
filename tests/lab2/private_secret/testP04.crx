// test all expression3

func main() : type {
  let a = not 5;
  let a = (4);
  let a = b[3];
  let a = ::foo();
  let a = 9;
  let a = 3.4;
  let a = true;
  let a = false;
}
