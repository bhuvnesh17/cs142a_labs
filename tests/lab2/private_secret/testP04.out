PROGRAM
  DECLARATION_LIST
    DECLARATION
      FUNCTION_DEFINITION
        PARAMETER_LIST
        TYPE
        STATEMENT_BLOCK
          STATEMENT_LIST
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        EXPRESSION3
                          LITERAL
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        EXPRESSION0
                          EXPRESSION1
                            EXPRESSION2
                              EXPRESSION3
                                LITERAL
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        DESIGNATOR
                          EXPRESSION0
                            EXPRESSION1
                              EXPRESSION2
                                EXPRESSION3
                                  LITERAL
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        CALL_EXPRESSION
                          EXPRESSION_LIST
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        LITERAL
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        LITERAL
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        LITERAL
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        LITERAL

